lib/browser/browser.ts

```diff
@@ -34,6 +34,78 @@ patchClass('FileReader');
 propertyPatch();
 registerElementPatch(_global);
 
+// Treat XMLHTTPRequest as a macrotask.
+patchXHR(_global);
```

Comment:
I wonder if it may be better to move this XHR patching to `XMLHttpRequest` since we already doing property patching there. Take a look and let me know what you think.